src/backend/access/table/table.c

```diff
@@ -227,7 +231,9 @@ CdbTryOpenTable(Oid relid, LOCKMODE reqmode, bool *lockUpgraded)
 		{
 			lockmode = RowExclusiveLock;
 			rel = try_table_open(relid, lockmode, false);
-			if (RelationIsAppendOptimized(rel))
```

Comment:
use the `RelationIsNonblockRelation` replace `RelationIsAppendOptimized` pls.

The `RelationIsNonblockRelation` contains the `PAX` am (which also not support the concurrently update or delete).